spark/src/test/java/org/apache/iceberg/spark/data/TestSparkParquetReader.java

```diff
@@ -67,4 +78,49 @@ protected void writeAndValidate(Schema schema) throws IOException {
       Assert.assertFalse("Should not have extra rows", rows.hasNext());
     }
   }
+
+  protected List<InternalRow> rowsFromFile(InputFile inputFile, Schema schema) throws IOException {
+    try (CloseableIterable<InternalRow> reader =
+        Parquet.read(inputFile)
+            .project(schema)
+            .createReaderFunc(type -> SparkParquetReaders.buildReader(schema, type))
+            .build()) {
+      return Lists.newArrayList(reader);
+    }
+  }
+
+  @Test
+  public void testInt96TimestampProducedBySparkIsReadCorrectly() throws IOException {
+    final SparkSession spark =
+        SparkSession.builder()
+            .master("local[2]")
+            .config("spark.sql.parquet.int96AsTimestamp", "false")
+            .getOrCreate();
```

Comment:
Is it possible to avoid creating a Spark session just to write a timestamp? What about calling Spark's `FileFormat` to write directly instead?

We wrap Spark's `FileFormat` in our DSv2 table implementation: https://github.com/Netflix/iceberg/blob/netflix-spark-2.4/metacat/src/main/java/com/netflix/iceberg/batch/BatchPatternWrite.java#L90

This test would run much faster by using that to create a file instead of creating a Spark context.

Reply:
Yes, I would much rather avoid creating a `SparkSession` here if possible. However, looking into [`ParquetFileFormat`](https://sourcegraph.com/github.com/apache/spark@d6a68e0b67ff7de58073c176dd097070e88ac831/-/blob/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala#L74:1) it seems like we would still need to pass a `SparkSession` to create the writer.

I can look at [`ParquetOutputWriter`](https://sourcegraph.com/github.com/apache/spark@d6a68e0b67ff7de58073c176dd097070e88ac831/-/blob/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOutputWriter.scala#L32) but I might need to match the configuration there with what Spark uses to write int96.

Reply:
Another approach would be to check-in a parquet file written by a spark and have the test just read it?

A drawback with that approach is that updating this file would be brittle, but I can check in the code that writes the file in an ignored test, but that should avoid us from creating a spark session during unit tests. What do you think @rdblue?

Reply:
At one point, we supported writing to Parquet using Spark's built-in ReadSupport. I think we can probably get that working again to create the files.

Reply:
Yes, looking at one of the tests we do support writing parquet files using Spark's WriteSupport.

 To be able to use a `FileAppender` I had to add a TimestampAsInt96 type (that can only be written using Spark's builtin WriteSupport) so that schema conversion within Iceberg's `ParquetWriteSupport` knows that this timestamps should be encoded as int96 in the  parquet  schema.